src/Kliva/Views/RecordPage.xaml.cs

```diff
@@ -41,14 +46,42 @@ private void OnViewModelPropertyChanged(object sender, System.ComponentModel.Pro
         {
             if (e.PropertyName.Equals(nameof(ViewModel.CurrentLocation), StringComparison.OrdinalIgnoreCase))
             {
-                TrackingMap.ClearMap<MapIcon>();
+                //ToDo PiNi: only do this the first time?
+                //and add button to re-center map?
+                //==> To discuss with team
+                CenterMap();
+
+                UpdateMapElements();
 
-                _currentLocation.Location = TrackingMap.Center = ViewModel.CurrentLocation;
-                TrackingMap.ZoomLevel = 16.0;
-                TrackingMap.MapElements.Add(_currentLocation);
+                if(!TrackingMap.MapElements.Contains(_currentLocation))
+                    TrackingMap.MapElements.Add(_currentLocation);
+
+                if (!TrackingMap.MapElements.Contains(_mapPolyline))
+                    TrackingMap.MapElements.Add(_mapPolyline);
             }
         }
 
+        private void UpdateMapElements()
+        {
+            List<BasicGeoposition> positions;
```

Comment:
I also doubted... But as this code is really simple and having the positions in the VM wouldn't give any added value for now, I thought it would be best to keep it simple and handle it in the view. But I agree on merging it now, and maybe re-discuss this later.